test/Microsoft.AspNet.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -234,5 +243,33 @@ public Task ExceptionBeforeFirstWriteIsReported()
             var ex = await Assert.ThrowsAsync<HttpRequestException>(() => response.Content.ReadAsStringAsync());
             Assert.IsType<InvalidOperationException>(ex.GetBaseException());
         }
+
+        private class DummyApplication : IHttpApplication<Context>
+        {
+            Func<Context, Task> _application;
+
+            public DummyApplication(Func<Context, Task> application)
```

Comment:
`ClientHandler` is hardcoded to accept only `IHttpApplication<Context>` so the `DummyApplication` here must be of that type.